sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/AttributesAdviceTest.java

```diff
@@ -34,78 +71,251 @@ void cleanup() {
             metric ->
-                assertThat(metric)
-                    .hasDoubleSumSatisfying(
-                        sum ->
-                            sum.hasPointsSatisfying(
-                                point ->
-                                    point.hasAttributesSatisfyingExactly(
-                                        equalTo(stringKey("key1"), "1"),
-                                        equalTo(stringKey("key2"), "2")))));
+                pointsAssert.hasPointSatisfying(
+                    assertThat(metric),
+                    point ->
+                        point.hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("key1"), "1"), equalTo(stringKey("key2"), "2"))));
   }
 
-  @Test
-  void counterWithAdviceAndViews() {
+  @ParameterizedTest
+  @ArgumentsSource(InstrumentsProvider.class)
+  void instrumentWithAdviceAndViews(
+      InstrumentFactory instrumentFactory, PointsAssert<AbstractPointAssert<?, ?>> pointsAssert) {
     InMemoryMetricReader reader = InMemoryMetricReader.create();
     meterProvider =
         SdkMeterProvider.builder()
             .registerMetricReader(reader)
             .registerView(
-                InstrumentSelector.builder().setType(InstrumentType.COUNTER).build(),
+                InstrumentSelector.builder().setName("test").build(),
                 View.builder()
                     .setAttributeFilter(key -> "key2".equals(key) || "key3".equals(key))
                     .build())
             .build();
 
-    DoubleCounterBuilder doubleCounterBuilder =
-        meterProvider.get("meter").counterBuilder("counter").ofDoubles();
-    ((ExtendedDoubleCounterBuilder) doubleCounterBuilder)
-        .setAdvice(advice -> advice.setAttributes(asList(stringKey("key1"), stringKey("key2"))));
-    DoubleCounter counter = doubleCounterBuilder.build();
-    counter.add(1, ATTRIBUTES);
+    Instrument instrument =
+        instrumentFactory.create(
+            meterProvider, "test", asList(stringKey("key1"), stringKey("key2")));
+    instrument.record(1, ATTRIBUTES);
 
     assertThat(reader.collectAllMetrics())
         .satisfiesExactly(
             metric ->
-                assertThat(metric)
-                    .hasDoubleSumSatisfying(
-                        sum ->
-                            sum.hasPointsSatisfying(
-                                point ->
-                                    point.hasAttributesSatisfyingExactly(
-                                        equalTo(stringKey("key2"), "2"),
-                                        equalTo(stringKey("key3"), "3")))));
+                pointsAssert.hasPointSatisfying(
+                    assertThat(metric),
+                    point ->
+                        point.hasAttributesSatisfyingExactly(
+                            equalTo(stringKey("key2"), "2"), equalTo(stringKey("key3"), "3"))));
+  }
+
+  static final class InstrumentsProvider implements ArgumentsProvider {
```

Comment:
The argument provider turned out to be a monster, but at least the test cases are very terse and easy to read